matrixctl/addons/adduser_jitsi/addon.py

```diff
@@ -50,28 +50,26 @@ def addon(arg: Namespace, yaml: YAML) -> int:
         Non-zero value indicates error code, or zero on success.
 
     """
-    address = (
-        yaml.get("server", "ssh", "address")
-        if yaml.get("server", "ssh", "address")
-        else f"matrix.{yaml.get('server', 'api', 'domain')}"
-    )
-    with SSH(
-        address,
-        yaml.get("server", "ssh", "user"),
-        yaml.get("server", "ssh", "port"),
-    ) as ssh:
```

Comment:
This is actually an old behavior which should have been removed by now. The code should look like:

```python
with SSH(
        yaml.get("server", "ssh", "address"),
        yaml.get("server", "ssh", "user"),
        yaml.get("server", "ssh", "port"),
    ) as ssh: ...
```

because new implementation of `yaml.get(...)` checks whether the requested config option exists and creates an error according to the missing option. It also exits the program. 
For example, when I remove `servers.default.ssh.address` from my config, it shows an error regarding the other option, to use the API domain. The output looks like:

```log
2022-12-16 02:38:23 matrixctl.handlers.yaml:381 [get] ERROR Please check your config file. For this operation your config file needs to have the entry "address" in "servers.default.ssh".
```

The user should make sure that `servers.[selected_server].ssh.address` exists, when they use a command, that needs a specific config option. If they want to avoid writing it multiple times, they could use jinja2 templates instead.

Therefore, you can always assume the returned config of `yaml.get()` might not be valid, but it exists. In many cases, the server itself checks, if a configured option is valid, so it is not actually checked in MatrixCtl to avoid the need to update the sanitizer each time the spec is changed. User, room, and event identifiers have simplified sanitizers because they can be mixed up fairly quickly and the error message from the server is less helpful.

Reply:
All `addon`s need to return an integer return value because their return value is returned to the OS as error code. They are dynamically imported based on the user's subcommand. This is done to accelerate the import process and more importantly, to give the user the option to use their own, maybe more specialized add-ons, not needed for most. (The latter is not implemented yet)

You find in `__main__.py` how it is returned, starting with this line:
https://github.com/MichaelSasser/matrixctl/blob/b32a5d19b1bfc32dcb994ffb1e11af47c3e89162/matrixctl/__main__.py#L183

I currently haven't worked out for which problem they return which error number. Later on, they should return different error numbers for different issues, which should make it simpler to use in bash-scripts.